### DISCUSSION TITLE
Gamepad type per application

### DISCUSSION THREAD

**Discussion by Bonaducci** [LizardByte/community#302] [Sunshine]
Sunshine allows emulating PS and Xbox gamepad inputs but the setting is global. Although it may seem reasonable to match this setting to the gamepad used by the client, some games may not support certain gamepad type forcing users to switch emulation to something else. Setting emulated gamepad per application would make it easier to configure the best type for the application in use.

Current workaround I'm using is emulating PS controller via Sunshine (as I'm using DualSense 5) and if the game does not support it out of the box, I'm launching the game through Steam with custom controller settings that will emulate xinput. This workaround works but forces me to add non-Steam games to Steam and configure controller settings separately from the application settings in Sunshine which is suboptimal.

Nice option would be to have the capability of overriding the emulated controller directly (eg. set global gamepad to PS and if the app does not support it, just override it per app).

**Comment by ld892012** [comment]
Maybe add possibility to change gamepad type directly into Moonlight can be a better way, some users launch their desktops or their game launchers (Playnite/GOG Galaxy) instead the game directly. 

And it's work if you start DS4Windows (or x360ce) as PreCommand for games who don't support PS ?

**Reply by davidspiess** [reply]
x360 setting is the best option for older games, which don't have native DS support, but in-game hints show the xbox controller layout. Newer games usually also support DS, so it would be nice to overwrite the global x360 setting with DS on application basis.

**Reply by Bonaducci** [reply]
That's pretty much what my original request says. This setup should be minimal with no additional apps, wrappers or other hacks. The only plausible roadblock I'd see is changing the controller emulation on the fly. I don't know if Sunshine does that on startup or on program launch.

**Comment by kurac338** [comment]
@ReenigneArcher any plans to implement this feature?
I believe it is a valid use case for PS controller users (DS4/DS5) when launching (older) apps that only support `XInput` natively.
Personally I believe a simplest implementation would be an option as a application [prep-command](https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/guides/app_examples.html#prep-commands) to tell Sunshine server to switch the emulation type (per app), but it needs to be supported with some CLI (flag) or ENV variable?

EDIT: after the reply I've found the [roadmap ticket](https://github.com/orgs/LizardByte/projects/13/views/1?sortedBy%5Bdirection%5D=desc&sortedBy%5BcolumnId%5D=Status&pane=issue&itemId=48745991)

**Reply by ReenigneArcher** [reply]
It's on the roadmap

**Comment by bell07** [comment]
What is current status for this feature request? The "roadmap ticket" link does not work anymore. :-(
My Use-Case is: I use Linux + Lutris + Wine  on server site, that does support xbox controller without tweaks, but does not support DS5 (Hidraw access is disabled by default in Wine).
Therefore the global setting "XOne" is fine for such games. But now I like to emulate some WII games, and like to use the gyro / IMU events from my PS4 controllers. Gyro events is working only with "DS5".

Another solution would be fine for me if the Gyro events are transferred to separate "Generic IMU" device in "XOne" mode.

**Reply by ReenigneArcher** [reply]
roadmap moved, here's the link https://github.com/LizardByte/roadmap/issues/31
